docs/modules/processor/pages/logging-api.adoc

```diff
@@ -0,0 +1,90 @@
+= Logging
+
+All warning and error messages in Asciidoctor.js are routed through a logger.
+Many of the messages also include context about the source location (file, dir, path, lineno), which can be useful for integrations and tooling.
+
+When using the API, you can feed your own logger to the `LoggerManager` to capture, route, or observe the messages.
+
+== Default Logger
+
+The default logger will output all warning and error messages to `stderr` (the standard error output stream).
+
+Depending on your JavaScript environment, `stderr` will be resolved to:
+
+* https://developer.mozilla.org/fr/docs/Web/API/Console/warn[`console.warn`] function in a browser environment
+* https://nodejs.org/api/process.html#process_process_stderr[`process.stderr`] property in a Node.js environment
+
+== In-memory Logger
+
+In addition to the default logger, Asciidoctor.js provides a built-in in-memory logger named `MemoryLogger`.
+This logger can be use to capture messages.
+
+First we need to instantiate this logger using the `$new()` function:
+
+```js
+const memoryLogger = asciidoctor.MemoryLogger.$new();
+```
+
+Once the logger is instantiated we can replace the default logger with the in-memory logger:
+
+```js
+asciidoctor.LoggerManager.setLogger(memoryLogger);
+```
+
+[NOTE]
+====
+If you want to restore the default logger, you should save a reference before replacing it:
+
+```js
+const loggerManager = asciidoctor.LoggerManager;
+const defaultLogger = loggerManager.getLogger(); // <1>
+try {
+  const memoryLogger = asciidoctor.MemoryLogger.$new();
+  loggerManager.setLogger(memoryLogger); // <2>
+  // do something with the in-memory logger
+} finally {
+  loggerManager.setLogger(defaultLogger); // <3>
+}
+```
+<1> save the default logger
+<2> replace the default logger with the in-memory logger
+<3> restore the default logger
+====
+
+=== Error and warning messages
+
+The in-memory logger stores every warning and error messages generated by the Asciidoctor.js processor.
+Once the processing is done, you can retrieve all these messages using the `getMessages` function:
```

Comment:
I was wondering if the messages are emitted on the fly as well, with events?

Reply:
Yes but you need to provide a custom logger or to extend a logger.

I will open another pull request with a dedicated API to create/extend logger.